src/circuits/joinsplit.tcc

```diff
@@ -21,35 +20,44 @@ template<
 class joinsplit_gadget : libsnark::gadget<FieldT>
 {
 private:
-    // Multipacking gadgets for the inputs (nullifierS, commitmentS, val_pub_in,
+    //  Multipacking gadgets for the inputs (nullifierS, commitmentS,
+    //  val_pub_in,
     // val_pub_out) (the root is a field element) `NumInputs + NumOutputs`
     // because we pack the nullifiers (Inputs of JS = NumInputs), the
     // commitments (Output of JS = NumOutputs) AND the v_pub_out taken out of
     // the mix (+1) AND the public value v_pub_in that is put into the mix (+1)
     // AND the signature hash h_sig (+1) AND the malleability tags h_iS (+
     // NumInputs)
+
     std::array<
         libsnark::pb_variable_array<FieldT>,
-        NumInputs + NumOutputs + 1 + 1 + 1 + NumInputs>
+        NumInputs + NumOutputs + 1 + NumInputs + 1>
         packed_inputs;
     std::array<
         libsnark::pb_variable_array<FieldT>,
-        NumInputs + NumOutputs + 1 + 1 + 1 + NumInputs>
+        NumInputs + NumOutputs + 1 + NumInputs + 1>
         unpacked_inputs;
+    // Residual bits from packing of 256-bit long string into 253-bit long field
+    // elements to which are added the public value of size 64 bits
+    const size_t residual_packing_bits =
```

Comment:
This variable does not represent bits, but rather a `number` of bits or the length of a bit string. Please rename accordingly